src/Std/Sat/CNF/Basic.lean

```diff
@@ -0,0 +1,188 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+prelude
+import Init.Data.List.Lemmas
+import Std.Sat.CNF.Literal
+
+namespace Std
+namespace Sat
+
+/--
+A clause in a CNF.
+
+The literal `(i, b)` is satisfied if the assignment to `i` agrees with `b`.
+-/
+abbrev CNF.Clause (α : Type u) : Type u := List (Literal α)
+
+/--
+A CNF formula.
+
+Literals are identified by members of `α`.
+-/
+abbrev CNF (α : Type u) : Type u := List (CNF.Clause α)
+
+namespace CNF
+
+/--
+Evaluating a `Clause` with respect to an assignment `f`.
+-/
+def Clause.eval (f : α → Bool) (c : Clause α) : Bool := c.any fun (i, n) => f i == n
+
+@[simp] theorem Clause.eval_nil (f : α → Bool) : Clause.eval f [] = false := rfl
+@[simp] theorem Clause.eval_succ (f : α → Bool) :
+    Clause.eval f (i :: c) = (f i.1 == i.2 || Clause.eval f c) := rfl
+
+/--
+Evaluating a `CNF` formula with respect to an assignment `f`.
+-/
+def eval (f : α → Bool) (g : CNF α) : Bool := g.all fun c => c.eval f
+
+@[simp] theorem eval_nil (f : α → Bool) : eval f [] = true := rfl
+@[simp] theorem eval_succ (f : α → Bool) : eval f (c :: g) = (c.eval f && eval f g) := rfl
+
+@[simp] theorem eval_append (f : α → Bool) (g h : CNF α) :
+    eval f (g ++ h) = (eval f g && eval f h) := List.all_append
+
+instance : HSat α (Clause α) where
+  eval assign clause := Clause.eval assign clause
+
+instance : HSat α (CNF α) where
+  eval assign cnf := eval assign cnf
+
+@[simp] theorem unsat_nil_iff_false : unsatisfiable α ([] : CNF α) ↔ False :=
```

Comment:
I think that the normal way to write this would just be
```suggestion
@[simp] theorem not_unsatisfiable_nil : ¬unsatisfiable α ([] : CNF α) :=
```
and `simp` should still be able to apply it.